Sources/Purchasing/Configuration.swift

```diff
@@ -131,12 +131,18 @@ import Foundation
          * Set `observerMode`.
          * - Parameter observerMode: Set this to `true` if you have your own IAP implementation and want to use only
          * RevenueCat's backend. Default is `false`.
-         *
-         * - Warning: This assumes your IAP implementation uses StoreKit 1.
-         * Observer mode is not compatible with StoreKit 2.
+         * - Parameter storeKitVersion: Set the StoreKit version you're using to make purchases.
+         * Default is `.storeKit1`.
          */
-        @objc public func with(observerMode: Bool) -> Configuration.Builder {
+        @objc public func with(
+            observerMode: Bool,
+            storeKitVersion: StoreKitVersion = .storeKit1
+        ) -> Configuration.Builder {
+            if self.storeKitVersion != StoreKitVersion.default {
+                Logger.warn(Strings.configure.observer_mode_with_storekit_version)
```

Comment:
Did you mean to use a different string here? This would tell you to use this method when using it with `.storeKit1` as well.

Reply:
Fixed, meant to use with `self.`, Basically this means: if the currently configured storeKitVersion is different than the default, it means the developer must have used `.with(sotreKitVersion:)` before calling this method.

Reply:
This is a smell that something is wrong with the API then.
Since we're making this a major release, I'd solve this issue by removing the default value for StoreKit version in this method. That way we don't need an easy to miss warning and instead developers know this at compile time.

Reply:
I think this makes sense! It indeed felt weird while implementing it. Didn't want to break the API for developers currently using StoreKit 1, but explicit is better than implicit.